### ISSUE TITLE
SourceLink Availability section test fails when local test assembly has no SourceLink data

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#675]
## Problem

`LibraryCommand_ExplicitSourceLinkAudit_IncludesSourceLinkAuditSection` expects `library <test-assembly> -S "Signals,SourceLink Availability,SourceLink Missing Files"` to render `## SourceLink Availability`.

On the current build, the local `dotnet-inspect.Tests.dll` PDB does not have usable SourceLink data, so only `## Signals` renders. `SourceLink Availability.CanRender(...)` stays false because the per-file source audit never runs (`HasSourceLink == false`, `TotalSourceFiles == 0`, `AllSourcesAccessible == null`).

## Repro

```bash
dotnet run --project src/dotnet-inspect.Tests -c Release -- \
  -method DotnetInspector.Tests.CommandExecutionTests.LibraryCommand_ExplicitSourceLinkAudit_IncludesSourceLinkAuditSection
```

Or directly:

```bash
dotnet run --project src/dotnet-inspect -c Release --no-build -- \
  library artifacts/bin/dotnet-inspect.Tests/release/dotnet-inspect.Tests.dll \
  -S "Signals,SourceLink Availability,SourceLink Missing Files" --markdown
```

Observed output contains `## Signals` but not `## SourceLink Availability`.

## Related wording issue

The `Signals` row can say `SourceLink | Not found` while the evidence says `SourceLink data found in PDB from standalone`. That appears to come from `FormatPdbEvidence(...)`; it is reporting PDB evidence, not SourceLink evidence, and is misleading when SourceLink is absent.

## Expected direction

Either:

1. Make the test use a fixture assembly/PDB with known SourceLink data, or
2. Change the test to assert the no-SourceLink behavior for the local test assembly.

Also consider tightening the `Signals` evidence wording so absent SourceLink does not say SourceLink data was found.
